### PR TITLE
refactor(vectorstore): introduce builder pattern for MilvusVectorStore

### PR DESCRIPTION
**Pull request by sobychacko**
- Replace configuration object with fluent builder pattern
- Move Milvus-related classes to dedicated milvus package
- Deprecate MilvusVectorStoreConfig in favor of builder
- Update constructor to use builder internally
- Maintain backward compatibility with deprecated config
- Add comprehensive builder methods with validation

### CODE REVIEW

File: spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/vectorstore/milvus/MilvusVectorStoreAutoConfiguration.java [hunk 1]
```diff
@@ -71,23 +71,13 @@ public MilvusVectorStore vectorStore(MilvusServiceClient milvusClient, Embedding
...
-			.withMetricType(MetricType.valueOf(properties.getMetricType().name()))
-			.withIndexParameters(properties.getIndexParameters())
-			.withEmbeddingDimension(properties.getEmbeddingDimension())
-			.withIDFieldName(properties.getIdFieldName())
-			.withAutoId(properties.isAutoId())
-			.withContentFieldName(properties.getContentFieldName())
-			.withMetadataFieldName(properties.getMetadataFieldName())
-			.withEmbeddingFieldName(properties.getEmbeddingFieldName())
+		return MilvusVectorStore.builder(milvusClient)
+			.embeddingModel(embeddingModel)
```
[the remark this answers is not in the document]
Comment (ilayaperumalg) [reply spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/vectorstore/milvus/MilvusVectorStoreAutoConfiguration.java:75]: We also need a test to validate this case in our AutoConfiguration IT for Milvus.

[the remark this answers is not in the document]
Comment (sobychacko) [reply spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/vectorstore/milvus/MilvusVectorStoreAutoConfiguration.java:75]: @ilayaperumalg Created this issue to track this: https://github.com/spring-projects/spring-ai/issues/1903. Maybe we can tackle this as a separate issue outside the scope of the builder refactoring?

File: vector-stores/spring-ai-milvus-store/src/main/java/org/springframework/ai/milvus/vectorstore/package-info.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Provides the API for embedding observations.
+ */
+@NonNullApi
```
Comment (markpollack) [inline vector-stores/spring-ai-milvus-store/src/main/java/org/springframework/ai/milvus/vectorstore/package-info.java:20]: not needed as they are in package-info

**Comment by markpollack** [comment]
merged in 218c9677f4fee043e365e8ebbe98b8bfbc81fbb4

**Pull request closed without merging**